client/stack-editor/lib/editor/index.coffee

```diff
@@ -220,6 +207,14 @@ module.exports = class StackEditorView extends kd.View
 
     @listenContentChanges()
 
+    if not @isMine and stackTemplate
+      @tabView.setClass 'StackEditorTabs isntMine'
+      @warningView.show()
+      @deleteStack.hide()
+      @saveButton.setClass 'isntMine'
+      @inputTitle.setClass 'template-title isntMine'
+      @titleActionsWrapper.hide()
```

Comment:
we can handle all these cases by defining same rule sets under `read-only` class of parent view.